src/orders/components/OrderGrantRefundPage/reducer.ts

```diff
@@ -20,47 +27,85 @@ export type GrantRefundAction =
       type: "setQuantity";
       lineId: string;
       amount: number;
+      unitPrice: number;
     }
   | {
       type: "setMaxQuantity";
-      lineIds: string[];
+      lines: Array<{
+        id: string;
+        quantity: number;
+        unitPrice: number;
+      }>;
     }
   | {
       type: "initState";
       state: GrantRefundState;
     }
   | {
       type: "toggleRefundShipping";
+    }
+  | {
+      type: "setRefundShipping";
+      refundShipping: boolean;
     };
 
 export const getGrantRefundReducerInitialState = (
   order: OrderDetailsGrantRefundFragment,
+  grantedRefund?: OrderDetailsGrantedRefundFragment,
 ): GrantRefundState => {
   const unfulfilledLines = order?.lines
     .filter(line => line.quantityToFulfill > 0)
-    .map<GrantRefundLineKeyValue>(line => [
-      line.id,
-      {
-        availableQuantity: line.quantity,
-        unitPrice: line.unitPrice.gross.amount,
-        selectedQuantity: 0,
-      },
-    ]);
+    .map<GrantRefundLineKeyValue>(line => {
+      const initialQuantity =
+        grantedRefund?.lines?.find(
+          initLine => (initLine as any).orderLine.id === line.id,
+        )?.quantity ?? 0;
+
+      return [
+        line.id,
+        {
+          isDirty: false,
+          availableQuantity: getLineAvailableQuantity(
+            line.id,
+            line.quantity,
+            order?.grantedRefunds,
+            grantedRefund?.id,
+          ),
+          unitPrice: line.unitPrice.gross.amount,
+          selectedQuantity: initialQuantity,
+          initialQuantity,
+        },
+      ];
+    });
```

Comment:
suggestion: this entire map function can be lifted up and named accordingly to what it does